packages/itwin/tree-widget/src/tree-widget-react/components/trees/models-tree/internal/ModelsTreeIdsCache.ts

```diff
@@ -105,6 +113,137 @@ export class ModelsTreeIdsCache implements Disposable {
     }
   }
 
+  private async *queryChildren({ elementIds }: { elementIds: Id64Array }): AsyncIterableIterator<{ id: Id64String; parentId: Id64String }> {
+    const ctes = [
+      `
+        ElementChildren(id, parentId) AS (
+          SELECT ECInstanceId id, Parent.Id parentId
+          FROM ${this.#hierarchyConfig.elementClassSpecification}
+          WHERE Parent.Id IN (${elementIds.join(", ")})
```

Comment:
what numbers of elements do we expect here? do we have any measures against a huge query that would result from a large number of ids?

Reply:
> function will be called with all grouped elements (should not be more than 1k unless hierarchy level limit is increased)

So it could easily be 10k, since we offer users to increase the loaded instances count to that by default. I think we need to batch these queries by ~1k. 10k ids here will produce a huge query.